### ISSUE TITLE
Add in bagging/subsampling option to sklearn style fit option

### ISSUE THREAD

**Issue by btcross26** [btcross26/genestboost#34]
**Is your feature request related to a problem? Please describe.**
Right now, bagging/subsampling can be done, but it would have to be in a convoluted manner using manual, for-loop style fitting.

**Describe the solution you'd like**
Add in a subsampling option to the boost method. This would go right before the model fit line, and predictions could still be performed on the full dataset.

**Describe alternatives you've considered**
Leave as-is - this is easy enough to add though and is a conventional option in other boosting implementations.

**Comment by btcross26** [comment]
Adding this to the boost method of the boosted model class so that it is also available for the surrogate loss function/loop boost approach.
